python/lsst/daf/butler/__init__.py

```diff
@@ -15,3 +15,7 @@
 from .registry import CollectionSearch, CollectionType, DatasetIdGenEnum, Registry, RegistryConfig
 from .transfers import YamlRepoExportBackend, YamlRepoImportBackend
 from .version import *
+
+# isort: off
+# _quantum_backed has to be imported after core and registry
+from ._quantum_backed import *
```

Comment:
Uh oh, does this mean we have an import cycle now?  If so, I guess we can live with it now, but we should refactor it away ASAP.

Reply:
It's not a cycle, I think it's because we use absolute imports (`from lsst.daf.butler import smthg`) in some of the modules. That needs correct ordering of imports, if I import them alphabetically then I'm getting an error:
```
$ python tests/test_quantumBackedButler.py
Traceback (most recent call last):
  File "tests/test_quantumBackedButler.py", line 26, in <module>
    from lsst.daf.butler import (
  File "/home/salnikov/gen3-middleware/daf_butler/python/lsst/daf/butler/__init__.py", line 12, in <module>
    from ._quantum_backed import *
  File "/home/salnikov/gen3-middleware/daf_butler/python/lsst/daf/butler/_quantum_backed.py", line 52, in <module>
    from .registry.bridge.monolithic import MonolithicDatastoreRegistryBridgeManager
  File "/home/salnikov/gen3-middleware/daf_butler/python/lsst/daf/butler/registry/bridge/monolithic.py", line 31, in <module>
    from lsst.daf.butler import NamedValueSet, StoredDatastoreItemInfo, ddl
ImportError: cannot import name 'NamedValueSet' from partially initialized module 'lsst.daf.butler' (most likely due to a circular import) (/home/salnikov/gen3-middleware/daf_butler/python/lsst/daf/butler/__init__.py)
```

I could try to replace some of the absolute imports with relative to resolve this.

Reply:
I fixed it by switching couple of modules in `registry/bridge` to use relative imports. There are many other places where we use absolute imports, maybe we should update all of them on a separate ticket.